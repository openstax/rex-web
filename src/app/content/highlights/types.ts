import { Highlight, HighlightColorEnum, HighlightsSummary, NewHighlight } from '@openstax/highlighter/dist/api';
import { ScrollTarget } from '../../navigation/types';
import { FiltersChange } from '../components/popUp/types';
import { LinkedArchiveTreeNode } from '../types';

export interface SummaryHighlights {
  [locationId: string]: {[pageId: string]: HighlightData[]};
}

export type OrderedSummaryHighlights = Array<{
  location: LinkedArchiveTreeNode,
  pages: Array<{
    pageId: string;
    highlights: HighlightData[];
  }>
}>;

export interface SummaryFilters {
  locationIds: string[];
  colors: HighlightColorEnum[];
}

export interface SummaryFiltersUpdate {
  locations?: FiltersChange<LinkedArchiveTreeNode>;
  colors?: FiltersChange<HighlightColorEnum>;
}

export type CountsPerSource = NonNullable<HighlightsSummary['countsPerSource']>;
export type HighlightColorCounts = CountsPerSource[string];

export type SummaryHighlightsPagination = null | {
  sourceIds: string[];
  page: number;
  perPage: number;
};

export interface State {
  currentPage: {
    pageId: string | null,
    highlights: null | HighlightData[];
    hasUnsavedHighlight: boolean;
    focused?: string;
  };
  summary: {
    open: boolean,
    pagination: SummaryHighlightsPagination,
    totalCountsPerPage: CountsPerSource | null;
    filters: SummaryFilters,
    loading: boolean;
    highlights: SummaryHighlights | null;
  };
}

export interface HighlightScrollTarget extends ScrollTarget {
  type: 'highlight';
  id: string;
}

// scopeId should not be nullable in the swagger
export type NewHighlightPayload = NewHighlight & {scopeId: string};
export type HighlightData = Highlight & {scopeId: string};
