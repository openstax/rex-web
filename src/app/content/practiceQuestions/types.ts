import { LinkedArchiveTreeSection } from '../types';

export interface State {
  summary: PracticeQuestionsSummary | null;
  selectedSection: LinkedArchiveTreeSection | null;
  currentQuestionIndex: number | null;
  questions: PracticeQuestion[];
  questionAnswers: QuestionAnswers;
  loading: boolean;
}

export interface PracticeQuestionsSummary {
  countsPerSource: {
    [key: string]: number;
  };
}

export interface PracticeAnswer {
  id: number;
  content_html: string;
  correctness: '0.0' | '1.0';
  feedback_html: string;
}

export interface PracticeQuestion {
  group_uuid: string;
  uid: string;
  stem_html: string;
  answers: PracticeAnswer[];
}

export type PracticeQuestions = PracticeQuestion[];

export interface QuestionAnswers {
  [key: string]: PracticeAnswer | null;
}

export interface SetPracticeAnswer {
  questionId: string;
  answer: PracticeAnswer | null;
}
