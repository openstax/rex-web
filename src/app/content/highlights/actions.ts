import { Highlight, NewHighlight, UpdateHighlightRequest } from '@openstax/highlighter/dist/api';
import { createStandardAction } from 'typesafe-actions';
import { LocationFilters } from '../components/popUp/types';
import {
  CountsPerSource,
  HighlightData,
  SummaryFilters,
  SummaryFiltersUpdate,
  SummaryHighlights,
  SummaryHighlightsPagination,
} from './types';

export const focusHighlight = createStandardAction('Content/Highlights/focus')<string>();
export const clearFocusedHighlight = createStandardAction('Content/Highlights/clear')();
export const createHighlight = createStandardAction('Content/Highlights/create')<NewHighlight, {
  revertingAfterFailure?: boolean,
  locationFilterId: string,
  pageId: string,
}>();
export const requestDeleteHighlight = createStandardAction('Content/Highlights/requestDelete')<Highlight, {
  locationFilterId: string,
  pageId: string,
}>();
export const receiveDeleteHighlight = createStandardAction('Content/Highlights/receiveDelete')<Highlight, {
  revertingAfterFailure?: boolean,
  locationFilterId: string,
  pageId: string,
}>();
export const updateHighlight = createStandardAction('Content/Highlights/update')<UpdateHighlightRequest, {
  revertingAfterFailure?: boolean,
  preUpdateData: UpdateHighlightRequest,
  locationFilterId: string,
  pageId: string,
}>();
export const receiveHighlights = createStandardAction(
  'Content/Highlights/receive'
)<{highlights: HighlightData[], pageId: string}>();
export const setAnnotationChangesPending = createStandardAction('Content/Highlights/setAnnotationChangesPending')<
  boolean
>();

export const openMyHighlights = createStandardAction('Content/Highlights/Summary/open')<void>();
export const closeMyHighlights = createStandardAction('Content/Highlights/Summary/close')<void>();
export const initializeMyHighlightsSummary = createStandardAction('Content/Highlights/Summary/init')<void>();

export const printSummaryHighlights = createStandardAction('Content/Highlights/Summary/print')();
export const toggleSummaryHighlightsLoading = createStandardAction('Content/Highlights/Summary/loading')<boolean>();

export const loadMoreSummaryHighlights = createStandardAction('Content/Highlights/Summary/loadMore')();
export const setSummaryFilters = createStandardAction('Content/Highlights/Summary/setFilters')<
  Partial<SummaryFilters>
>();
export const updateSummaryFilters = createStandardAction('Content/Highlights/Summary/updateFilters')<
  Partial<SummaryFiltersUpdate>
>();
export const receiveSummaryHighlights = createStandardAction('Content/Highlights/Summary/receiveHighlights')<
  SummaryHighlights,
  {
    pagination: SummaryHighlightsPagination,
    filters?: SummaryFilters,
    isStillLoading?: boolean
  }
>();
export const receiveHighlightsTotalCounts = createStandardAction(
  'Content/receiveHighlightsTotalCounts'
)<CountsPerSource, LocationFilters>();
