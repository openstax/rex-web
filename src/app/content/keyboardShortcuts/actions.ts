import { createStandardAction } from 'typesafe-actions';

export const openKeyboardShortcutsMenu = createStandardAction('Content/KeyboardShortcuts/Menu/open')<void>();
export const closeKeyboardShortcutsMenu = createStandardAction('Content/KeyboardShortcuts/Menu/close')<void>();
