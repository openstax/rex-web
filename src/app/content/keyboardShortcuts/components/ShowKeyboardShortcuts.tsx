import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled, { css } from 'styled-components/macro';
import { htmlMessage } from '../../../components/htmlMessage';
import theme from '../../../theme';
import { PopupBody } from '../../styles/PopupStyles';

// tslint:disable-next-line:variable-name
const ShowKeyboardShortcutsBody = styled(PopupBody)`
  background-color: ${theme.color.neutral.darker};
  color: ${theme.color.text.default};
  font-size: 1.6rem;
  line-height: 2.5rem;
  padding: 2rem 3.2rem 3.2rem 3.2rem;
  ${theme.breakpoints.mobile(css`
    padding: 1.6rem;
    text-align: left;
  `)}
`;

// tslint:disable-next-line: variable-name
const ShortcutsHeadingDiv = styled.div`
  font-size: 1.8rem;
  font-weight: bold;
  line-height: 3rem;
  ${theme.breakpoints.mobile(css`
    font-size: 1.6rem;
    line-height: 2.5rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const ShortcutsHeading = ({msgKey}: {msgKey: string}) => (
  <ShortcutsHeadingDiv>
    <FormattedMessage id={`i18n:a11y:keyboard-shortcuts:${msgKey}`}>
      {(msg) => msg}
    </FormattedMessage>
  </ShortcutsHeadingDiv>
);

// tslint:disable-next-line: variable-name
const ShortcutsCard = styled.div`
  background-color: ${theme.color.white};
  border: 1px solid ${theme.color.neutral.darkest};
  margin: 2rem 0;
  padding: 1.6rem 0;
  width: 100%;
  ${theme.breakpoints.mobile(css`
    margin: 1.6rem 0;
    padding: 0.8rem;
  `)}
`;

// tslint:disable-next-line: variable-name
const ShortcutsTable = styled.div`
  border-collapse: separate;
  border-spacing: 3.2rem 1.6rem;
  display: table;
  ${theme.breakpoints.mobile(css`
    display: block;
  `)}
`;

// tslint:disable-next-line: variable-name
const ShortcutRow = styled.div`
  display: table-row;
  ${theme.breakpoints.mobile(css`
    display: block;
  `)}
`;

// tslint:disable-next-line: variable-name
const ShortcutBlock = styled.div`
  display: table-cell;
  min-width: 16rem;
  vertical-align: middle;
  ${theme.breakpoints.mobile(css`
    display: block;
    margin: 0.8rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const ShortcutKey = styled.span`
  background-color: ${theme.color.neutral.darker};
  border: 1px solid ${theme.color.neutral.formBorder};
  border-radius: 3px;
  display: inline-block;
  font-weight: bold;
  padding: 0.4rem 1.2rem 0.3rem;
  vertical-align: middle;
`;

// tslint:disable-next-line: variable-name
export const Shortcut = ({keys, msgKey}: {keys: string[], msgKey: string}) => (
  <ShortcutRow>
    <ShortcutBlock>
      {keys.map<React.ReactNode>(
        (k, index) => <ShortcutKey key={index}>
          <FormattedMessage id={`i18n:a11y:keyboard-shortcuts:keys:${k}`}>
            {(msg) => msg}
          </FormattedMessage>
        </ShortcutKey>
      ).reduce((prev, curr) => [prev, ' + ', curr])}
    </ShortcutBlock>

    <ShortcutBlock>
      <FormattedMessage id={`i18n:a11y:keyboard-shortcuts:${msgKey}`}>
        {(msg) => msg}
      </FormattedMessage>
    </ShortcutBlock>
  </ShortcutRow>
);

// tslint:disable-next-line: variable-name
const CaretMessageDiv = styled.div`
  /* Nothing here at the moment */
`;

// tslint:disable-next-line: variable-name
export const CaretMessage = htmlMessage(
  'i18n:a11y:keyboard-shortcuts:caret-extension', CaretMessageDiv
);

// tslint:disable-next-line: variable-name
const ShowKeyboardShortcuts = () => (
  <ShowKeyboardShortcutsBody
    data-testid='show-keyboard-shortcuts-body'
    data-analytics-region='KS popup'
  >
    <ShortcutsHeading msgKey='sub-heading'/>

    <ShortcutsCard>
      <ShortcutsTable>
        <Shortcut keys={['shift', '?']} msgKey='open-menu'/>
        <Shortcut keys={['tab']} msgKey='move-forward'/>
        <Shortcut keys={['shift', 'tab']} msgKey='move-backward'/>
        <Shortcut keys={['h']} msgKey='move-focus-in-out'/>
        <Shortcut keys={['tab']} msgKey='move-through-note-editing'/>
        <Shortcut keys={['space']} msgKey='select-highlight-color'/>
        <Shortcut keys={['enter']} msgKey='save-or-cancel'/>
      </ShortcutsTable>
    </ShortcutsCard>

    <ShortcutsHeading msgKey='creating-highlights-and-notes'/>

    <CaretMessage/>

    <ShortcutsCard>
      <ShortcutsTable>
        <Shortcut keys={['arrows']} msgKey='move-forward'/>
        <Shortcut keys={['shift', 'arrows']} msgKey='move-backward'/>
        <Shortcut keys={['h']} msgKey='move-focus-in-out'/>
        <Shortcut keys={['tab']} msgKey='move-through-note-editing'/>
        <Shortcut keys={['space']} msgKey='select-highlight-color'/>
        <Shortcut keys={['enter']} msgKey='save-or-cancel'/>
      </ShortcutsTable>
    </ShortcutsCard>
  </ShowKeyboardShortcutsBody>
);

export default ShowKeyboardShortcuts;
