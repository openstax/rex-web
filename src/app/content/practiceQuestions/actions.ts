import { createStandardAction } from 'typesafe-actions';
import { LinkedArchiveTreeSection } from '../types';
import { PracticeQuestion, PracticeQuestionsSummary, SetPracticeAnswer } from './types';

export const setSelectedSection = createStandardAction(
  'Content/PracticeQuestions/setSelectedSection'
)<LinkedArchiveTreeSection | null>();
export const setQuestions = createStandardAction('Content/PracticeQuestions/setQuestions')<PracticeQuestion[]>();
export const nextQuestion = createStandardAction('Content/PracticeQuestions/nextQuestion')<void>();
export const setAnswer = createStandardAction('Content/PracticeQuestions/setAnswer')<SetPracticeAnswer>();
export const finishQuestions = createStandardAction('Content/PracticeQuestions/finishQuestions')<void>();

export const receivePracticeQuestionsSummary = createStandardAction(
  'Content/PracticeQuestions/Summary/receive'
)<PracticeQuestionsSummary>();

export const openPracticeQuestions = createStandardAction('Content/PracticeQuestions/open')<void>();
export const closePracticeQuestions = createStandardAction('Content/PracticeQuestions/close')<void>();
