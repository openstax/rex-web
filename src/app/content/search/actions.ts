import { createAction, createStandardAction } from 'typesafe-actions';
import { SearchScrollTarget, SelectedResult, State } from './types';

export const receiveSearchResults = createAction('Content/Search/receiveResults', (action) =>
  (
    results: Exclude<State['results'], null>,
    meta?: {searchScrollTarget?: SearchScrollTarget}
  ) => action(results, meta)
);

export const requestSearch = createAction('Content/Search/request', (action) =>
  (
    query: string,
    meta?: {
      isResultReload: boolean,
      searchScrollTarget?: SearchScrollTarget,
    }
  ) => action(query, meta)
);

export const clearSearch = createStandardAction('Content/Search/clear')();
export const openMobileToolbar = createStandardAction('Content/Search/openMobileToolbar')();
export const openSearchResultsMobile = createStandardAction('Content/Search/open')();
export const closeSearchResultsMobile = createStandardAction('Content/Search/close')();

export const selectSearchResult = createStandardAction('Content/Search/selectResult')<SelectedResult>();

export const openSearchInSidebar = createStandardAction('Content/Search/openSearchInSidebar')();
