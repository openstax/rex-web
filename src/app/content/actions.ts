import { createStandardAction } from 'typesafe-actions';
import * as highlightingActions from './highlights/actions';
import * as practiceQuestionsActions from './practiceQuestions/actions';
import * as searchActions from './search/actions';
import * as studyGuidesActions from './studyGuides/actions';
import { ArchivePage, Book, Params, State } from './types';

export const search = searchActions;
export const highlighting = highlightingActions;
export const studyGuides = studyGuidesActions;
export const practiceQuestions = practiceQuestionsActions;

export const openToc = createStandardAction('Content/openToc')<void>();
export const closeToc = createStandardAction('Content/closeToc')<void>();
export const resetToc = createStandardAction('Content/resetToc')<void>();

export const receiveBuyPrintConfig = createStandardAction('Content/receiveBuyPrintConfig')<State['buyPrint']>();

export const requestBook = createStandardAction('Content/requestBook')<Params['book']>();
export const receiveBook = createStandardAction('Content/receiveBook')<Book>();

export const requestPage = createStandardAction('Content/requestPage')<Params['page']>();
export const receivePage = createStandardAction('Content/receivePage')<
  ArchivePage & {references: State['references']}
>();
export const receivePageNotFoundId = createStandardAction('Content/receivePageNotFoundId')<string>();

export const openNudgeStudyTools = createStandardAction('Content/openNudgeStudyTools')();
export const closeNudgeStudyTools = createStandardAction('Content/closeNudgeStudyTools')();
