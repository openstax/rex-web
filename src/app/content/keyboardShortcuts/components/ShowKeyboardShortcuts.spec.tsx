import React from 'react';
import renderer from 'react-test-renderer';
import createTestStore from '../../../../test/createTestStore';
import { book } from '../../../../test/mocks/archiveLoader';
import TestContainer from '../../../../test/TestContainer';
import { Store } from '../../../types';
import { receiveBook } from '../../actions';
import ShowKeyboardShortcuts from './ShowKeyboardShortcuts';

describe('ShowKeyboardShortcuts', () => {
  let store: Store;
  let render: () => JSX.Element;

  beforeEach(() => {
    store = createTestStore();
    render = () => <TestContainer store={store}>
      <ShowKeyboardShortcuts />
    </TestContainer>;
  });

  it('renders the keyboard shortcuts menu', () => {
    store.dispatch(receiveBook(book));

    const component = renderer.create(render());

    expect(component.toJSON()).toMatchSnapshot();
  });
});
