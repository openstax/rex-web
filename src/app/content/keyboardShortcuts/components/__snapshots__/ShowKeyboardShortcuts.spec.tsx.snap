// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowKeyboardShortcuts renders the keyboard shortcuts menu 1`] = `
<div
  className="PopupStyles__PopupBody-w0z1hl-2 ShowKeyboardShortcuts__ShowKeyboardShortcutsBody-sc-12dalxo-0 eGlfQX"
  data-analytics-region="KS popup"
  data-testid="show-keyboard-shortcuts-body"
>
  <div
    className="ShowKeyboardShortcuts__ShortcutsHeadingDiv-sc-12dalxo-1 pcTDQ"
  >
    Using a Keyboard for Navigation
  </div>
  <div
    className="ShowKeyboardShortcuts__ShortcutsCard-sc-12dalxo-2 goyzXO"
  >
    <div
      className="ShowKeyboardShortcuts__ShortcutsTable-sc-12dalxo-3 grHFdD"
    >
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Shift
          </span>
           + 
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            ?
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Pull up keyboard shortcuts menu
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Tab
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move forward through book page elements, highlights and notes
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Shift
          </span>
           + 
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Tab
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move backward through book page elements, highlights and notes
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            h
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move focus into and out of a note
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Tab
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move through note editing options
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Space
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Select a different highlight color
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Enter
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Save or Cancel note edits
        </div>
      </div>
    </div>
  </div>
  <div
    className="ShowKeyboardShortcuts__ShortcutsHeadingDiv-sc-12dalxo-1 pcTDQ"
  >
    Creating Highlights and Notes with a Keyboard
  </div>
  <div
    className="ShowKeyboardShortcuts__CaretMessageDiv-sc-12dalxo-7 dCQGYV"
    dangerouslySetInnerHTML={
      Object {
        "__html": "This functionality requires the use of a <strong>Caret Browsing extension</strong>, which enables you to browse and select the text of web pages using arrow keys.",
      }
    }
  />
  <div
    className="ShowKeyboardShortcuts__ShortcutsCard-sc-12dalxo-2 goyzXO"
  >
    <div
      className="ShowKeyboardShortcuts__ShortcutsTable-sc-12dalxo-3 grHFdD"
    >
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Arrows
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move forward through book page elements, highlights and notes
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Shift
          </span>
           + 
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Arrows
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move backward through book page elements, highlights and notes
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            h
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move focus into and out of a note
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Tab
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Move through note editing options
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Space
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Select a different highlight color
        </div>
      </div>
      <div
        className="ShowKeyboardShortcuts__ShortcutRow-sc-12dalxo-4 iaCHkf"
      >
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          <span
            className="ShowKeyboardShortcuts__ShortcutKey-sc-12dalxo-6 kUZwXn"
          >
            Enter
          </span>
        </div>
        <div
          className="ShowKeyboardShortcuts__ShortcutBlock-sc-12dalxo-5 fkJiLz"
        >
          Save or Cancel note edits
        </div>
      </div>
    </div>
  </div>
</div>
`;
